File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/C/s378126458_comments_removed.c
#include <stdio.h>
#include <string.h>

int main() {
    int N, Q;
    scanf("%d %d", &N, &Q);
    char s[N+1];
    scanf("%s", s);
    int pre[N+1];
    memset(pre, 0, sizeof(pre));

    for(int i = 1; i < N; i++) {
        if(s[i-1] == 'A' && s[i] == 'C') {
            pre[i] = 1;
        }
    }

    for(int i = 1; i < N; i++) {
        pre[i] += pre[i-1];
    }

    while(Q-- > 0) {
        int l, r;
        scanf("%d %d", &l, &r);
        r--;
        printf("%d\n", pre[r] - pre[l-1]);
    }
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/mad/Java/Go/s378126458_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)

	sc.Scan()
	N, _ := strconv.Atoi(sc.Text())
	sc.Scan()
	Q, _ := strconv.Atoi(sc.Text())
	sc.Scan()
	s := sc.Text()

	pre := make([]int, N+1)
	for i := 1; i < N; i++ {
		if s[i-1] == 'A' && s[i] == 'C' {
			pre[i] = 1
		}
	}

	for i := 1; i < N; i++ {
		pre[i] += pre[i-1]
	}

	for Q > 0 {
		sc.Scan()
		l, _ := strconv.Atoi(sc.Text())
		sc.Scan()
		r, _ := strconv.Atoi(sc.Text())
		r--
		fmt.Println(pre[r] - pre[l-1])
		Q--
	}
}
// End of Code
